src/mito2/src/region.rs

```diff
@@ -297,19 +298,30 @@ impl MitoRegion {
     }
 
     /// Sets the staging state.
+    ///
     /// You should call this method in the worker loop.
     /// Transitions from Writable to Staging state.
-    pub(crate) fn set_staging(&self) -> Result<()> {
+    /// Cleans any existing staging manifests before entering staging mode.
+    pub(crate) async fn set_staging<'a, 'b>(
+        &'a self,
+        manager: RwLockWriteGuard<'b, RegionManifestManager>,
+    ) -> Result<()>
+    where
+        'a: 'b,
+    {
+        manager.store().clear_staging_manifests().await?;
```

Comment:
If we call this before setting staging, maybe we can use remove_all?
https://docs.rs/opendal/latest/opendal/struct.Operator.html#method.remove_all